src/transformers/modeling_utils.py

```diff
@@ -1993,3 +1994,23 @@ def forward(self, hidden_states):
         return torch.cat(output_chunks, dim=chunk_dim)
 
     return forward_fn(*input_tensors)
+
+
+def align_dataset_labels_with_config(
+    dataset: Dataset, config: PretrainedConfig, label_column: str = "labels"
+) -> Dataset:
+    def _process_labels(batch):
+        int2str_function = dataset.features[label_column].int2str
+        label_name_to_id = {k.lower(): v for k, v in config.label2id.items()}
+        label_names = int2str_function(batch[label_column])
+        # TODO(lewtun): we need to also lowercase dataset label names?
+        batch[label_column] = label_name_to_id[label_names]
+        return batch
+
+    dataset = dataset.map(_process_labels)
```

Comment:
You can pass `features=` directly in `map` instead of casting afterwards.

Also some minor optimization stuff:
- please use a batched map, this will be faster
- compute `label_name_to_id` and `int2str_function` outside the `_process_labels` function